src/Log/Engine/SyslogLog.php

```diff
@@ -81,6 +86,18 @@ class SyslogLog extends BaseLog
      */
     protected $_open = false;
 
+    /**
+     * @inheritDoc
+     */
+    public function __construct(array $config = [])
+    {
+        parent::__construct($config);
+
+        if ($this->formatter && !isset($this->_config['formatter']['format'])) {
+            $this->formatter->setConfig('format', $this->_config['format']);
```

Comment:
Can't setting required config for the formatter be done at instance construction itself?

Reply:
There aren't shared/common configs between formatters so BaseLog doesn't know what to set.

Reply:
Ah, I missed the fact that the log class' option is being set as the formatter's option. Still how about modifying `$config['formatter']` before calling `parent::__construct()`

Reply:
I was having trouble making it 100% BC. Custom loggers could override the `_defaultConfig` value, so I was using the result of `setConfig()` in `BaseLog::__construct()` to resolve that, then I looked to see it the value was set.

However, I'd like to remove the formatter-specific options from the log config in `_defaultConfig` and deprecate when it is set.